pkg/parser/parser.peg

```diff
@@ -316,53 +316,82 @@ InlineAttributes <- "[" attrs:(GenericAttribute)* "]" {
-// considered, but it must not start with a # or dot.  Anything after a comma
-// (including the comma) is discarded.  (Future support for named attributes?)
-QuotedTextRole <- "[" role:QuotedTextRoleWord? "]" {
+// asciidoctor is inconsistent in handling quoted text attributes.
+// It assumes that unless the element leads with a short-hand, the whole thing is the role.
+// This is probably unintentional inconsistency, and one we've not elected to match.
+QuotedTextAttrs <- "[" attrs:(QuotedTextAttrRole? ShortHandAttr* NamedAttr*) "]" {
+    return types.NewElementAttributes(attrs)
+}
+
+QuotedTextAttrRole <- role:PositionalValue {
     return types.NewElementRole(role.(string))
 }
 
-QuotedTextRoleWord <- ![#.] role:([^\]]* {
-    return strings.SplitN(string(c.text), ",", 2)[0], nil
-}) {
-    return role, nil
+StandaloneAttributes <- attributes:(ElementAttribute)+ BlankLine* EOF { // standalone attributes, i.e., with nothing afterwards
+    return types.NewAttributes(attributes)
 }
 
-// This is for [.role#id] (or [#id.role]).  Multiple roles can be present.  If multiple
-// ids are present only the first is retained.  Either the role or id may be omitted,
-// but at least one must be present.
-QuotedTextShortHand <- "[" attrs:( QuotedTextShortRole / QuotedTextShortID )+ "]" {
-    return attrs, nil
+ShortHandAttr <- ShortHandAttrID / ShortHandAttrOption / ShortHandAttrRole
+
+ShortHandAttrOption <- "%" option:ShortHandValue {
+    return types.NewElementOption(option.(string))
 }
 
-QuotedTextShortRole <- '.' role:([^.#\]]+ { 
-        return string(c.text), nil 
-    }) {
+ShortHandAttrID <- "#" id:ShortHandValue {
+    return types.NewElementID(id.(string))
+}
+
+ShortHandAttrRole <- '.' role:ShortHandValue {
     return types.NewElementRole(role.(string))
 }
 
-QuotedTextShortID <- '#' id:([^.#\]]+ {
-        return string(c.text), nil
-    }) {
-    return types.NewElementID(id.(string))
+// PositionalValue is an unnamed attribute.
+PositionalValue <- ShortHandValue
+
+ShortHandValue <- ShortHandValuePlain / AttrValueSingleQuoted / AttrValueDoubleQuoted
+
+// ShortHandValuePlain is sort of like AttrValuePlain, but it also needs to exclude the characters
+// used to start a short hand role, id, or option, as well as equals signs.
+ShortHandValuePlain <- [^,\r\n"' \t.#%=\]]+ {
+    return string(c.text), nil
 }
 
-StandaloneAttributes <- attributes:(ElementAttribute)+ BlankLine* EOF { // standalone attributes, i.e., with nothing afterwards
-    return types.NewAttributes(attributes)
+NamedAttr <- ","? Space* key:NamedAttrKey Space* "=" Space* value:NamedAttrValue Space* {
+    return types.NewElementNamedAttr(key.(string), value.(string))
+}
+
+// The spec says attributes have be alphanumeric but does not consider foreign letters.  We are more generous.
+NamedAttrKey <- [\pL0-9_][\pL0-9_-]* {
+    return string(c.text), nil
+}
+
+// A few words about attribute values.  The official documentation says that they have to be quoted if they
```

Comment:
what official documentation are you referring too? Asciidoc or Asciidoctor?

Reply:
asciidoctor.

I am however following up on this -- because I wound up relexing this too -- we had too many tests that broke when I applied this (later commit) to more blocks.

I'll post an update here -- stay tuned.

Reply:
I do think I'll stop commenting on the grammar in comments though.  I generally would like to document the choices made, but it's almost impossible to do that without discussing asciidoctor, and without pointing out some of it's flaws where we differ.

I think it's probably better to have this discussion in the WG when we start working on a formal grammar.

Reply:
ok, sounds good to me. Let's stay neutral here but let's raise all those issues in the WG, then. But please don't feel offended by my request here!
Maybe we should start a doc somewhere to collect them as we keep working on the grammar? WDYT?

Reply:
Probably... I'm just not sure where to do that -- normally I'd think git would be the best place to do that (and even in the grammar spec itself), but I also understand your concerns.

The WG isn't going yet, they're still working on "governance", and I think it may be a while yet before it starts doing anything productive.

I just hope Dan & co. can refrain from adding "new features" (syntactic sugar especially) in the mean time.